### ISSUE TITLE
Understading train_collate_fn

### ISSUE THREAD

**Issue by CodingTomo** [mfarre/Video-LLaVA-7B-hf-CinePile#5]
Hi @mfarre!
Great work for the community!

I'm trying to fine tune VideoLLaVA on a driving domain and I took your notebook as starting point, but i'm struggling to understand what exactly happens here:

```
  batch = processor(text=texts, videos=videos, padding=True,  truncation = True, max_length=MAX_LENGTH_PROCESSOR, return_tensors="pt")
  labels = batch["input_ids"].clone()
  labels[labels == processor.tokenizer.pad_token_id] = -100
  batch["labels"] = labels
  input_ids = batch["input_ids"]
  attention_mask = batch["attention_mask"]
  pixel_values_videos = batch["pixel_values_videos"]
  labels = batch["labels"]
```

I'm working with prompt like: 
`USER: <video>\n{question} ASSISTANT: {answer}`

After apply the `processor` in `input_ids` at the beginning I have a bunch of tokens representing `USER:` followed by all the video tokens (`32001`) and finally the tokens associated to  `\n{question} ASSISTANT: {answer}`.

I'm assuming that this is the correct set up. However, I get lost on:
```labels[labels == processor.tokenizer.pad_token_id] = -100```

- What this line of code should do? 
- The value of `processor.tokenizer.pad_token_id` is `32002` (different from the video token). Is that correct?
- Finally `labels` in the should contain just the answer or it is correct that contains all the prompt? 

Thanks!

**Comment by mfarre** [comment]
hello @CodingTomo, great project, let me know how it goes!

The line that you are mentioning is basically there to not take into account padding tokens for the loss calculation. This is necessary when dealing with batch processing, where shorter sequences are padded to match the longest sequence in the batch. This would matter if you increase batch size to more than one.

is correct that labels contains the full prompt but recently during our trainings of SmolVLM2 https://huggingface.co/HuggingFaceTB/SmolVLM2-2.2B-Instruct we validated that masking user prompts (the question) helps a bit when we check the results in benchmarks.

**Comment by CodingTomo** [comment]
Thanks @mfarre  I got it, I was debugging with `batchsize=1` :)

Thanks for the suggestion! I will definitely look into SmolVLM2. I'm evaluating solutions for captioning videos taken from on-board vehicle cameras. The results I obtained with zero-shot models are very bad when asking for detailed descriptions of the ego-vehicle's actions and dynamics relative to other vehicles. I believe that for this particular domain, I need at least a LoRA fine-tuning.

**Comment by mfarre** [comment]
Check this out @CodingTomo https://github.com/merveenoyan/smol-vision/blob/main/Fine_tune_SmolVLM2_on_Video.ipynb
